server/deploys_resource.go

```diff
@@ -79,11 +78,16 @@ func (dr *DeploysResource) Create(u *url.URL, h http.Header, req *DeployRequest)
 			log.Printf("Can't destroy previous versions (%d previous versions), disabling destroy.", len(versions))
 			req.Deploy.DestroyPrevious = false
 		} else {
-			go dr.destroyPrevious(serviceName, versions[0], req.Deploy.Version)
+			go dr.destroyPrevious(serviceName, versions[0], req.Deploy.Version, destroyPreviousCheckDelay)
 		}
 	}
 
-	_, err := dr.Fleet.StartUnit(fleetName, options)
+	unit := schema.Unit{
```

Comment:
This used to be in the client package, but is now pushed out to here.  I think this is okay, but I can probably condense this line.  The extra labels probably aren't necessary given the few struct arguments being well named.